registry/search.go

```diff
@@ -0,0 +1,139 @@
+package registry // import "github.com/docker/docker/registry"
+
+import (
+	"context"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/api/types/registry"
+	"github.com/docker/docker/dockerversion"
```

Comment:
This import causes the CLI to also pull in https://github.com/docker/cli/pull/4010/commits/a5d122d15c095955d04136348e4c693a76b24613 (commit vendoring is from a temporary branch, so link will likely be broken at some point);

- github.com/docker/docker/dockerversion
    - -> github.com/docker/docker/pkg/useragent
    - -> github.com/docker/docker/pkg/parsers/kernel
    - -> golang.org/x/sys/windows/registry

We'll probably need to have a look at passing the `dockerversion` / `dockerversion.DockerUserAgent(ctx)` somehow

/cc @corhere @neersighted

Reply:
Ah, shoot, interesting.

Reply:
Quick search for that;

When Authenticating;
https://github.com/moby/moby/blob/a082bbcb30c3aafd58cbec5023ac1fdb57161348/daemon/auth.go#L11-L13

Distribution sets it up as part of the transport;
https://github.com/moby/moby/blob/a082bbcb30c3aafd58cbec5023ac1fdb57161348/distribution/registry.go#L100-L101

And for plugins;
https://github.com/moby/moby/blob/a082bbcb30c3aafd58cbec5023ac1fdb57161348/plugin/registry.go#L30-L34

So looks like it's a bit "all over the place"